include/xsparse/level_capabilities/co_iteration.hpp

```diff
@@ -0,0 +1,197 @@
+                                           { return std::tuple(args.iter_helper(i, pkm1)...); },
+                                           coiterate.m_levelsTuple))
+            {
+            }
+
+            class iterator
+            {
+            private:
+                coiteration_helper const& m_coiterHelper;
+                std::tuple<typename Levels::LevelCapabilities::iteration_helper::iterator...>
+                    iterators;
+                IK min_ik;
+
+            private:
+                template <typename... T1, typename... T2>
+                inline constexpr auto compareHelper(const std::tuple<T1...>& t1,
+                                                    const std::tuple<T2...>& t2) const noexcept
+                {
+                    static_assert(sizeof...(T1) == sizeof...(T2));
+
+                    return compare(t1, t2, std::make_index_sequence<sizeof...(T1)>{});
+                }
+
+                template <typename... T1, typename... T2, std::size_t... I>
+                inline constexpr auto compare(const std::tuple<T1...>& t1,
+                                              const std::tuple<T2...>& t2,
+                                              std::index_sequence<I...>) const noexcept
+                {
+                    return std::tuple{ std::get<I>(t1) == std::get<I>(t2)... };
+                }
+
+                template <typename... T1, typename... T2>
+                inline constexpr void min_helper(const std::tuple<T1...>& t1,
+                                                 const std::tuple<T2...>& t2)
+                {
+                    static_assert(sizeof...(T1) == sizeof...(T2));
+
+                    calc_min_ik(t1, t2, std::make_index_sequence<sizeof...(T1)>{});
+                }
+
+                template <typename... T1, typename... T2, std::size_t... I>
+                inline constexpr void calc_min_ik(const std::tuple<T1...>& t1,
+                                                  const std::tuple<T2...>& t2,
+                                                  std::index_sequence<I...>)
+                {
+                    min_ik = std::min({ (std::get<I>(t1) != std::get<I>(t2))
+                                            ? std::get<0>(*std::get<I>(t1))
+                                            : min_ik... });
+                }
+
+            public:
+                using iterator_category = std::forward_iterator_tag;
+                using reference = typename std::
+                    tuple<IK, std::tuple<std::optional<typename Levels::BaseTraits::PK>...>>;
+
+                explicit inline iterator(
+                    coiteration_helper const& coiterHelper,
+                    std::tuple<typename Levels::LevelCapabilities::iteration_helper::iterator...>
+                        it) noexcept
+                    : m_coiterHelper(coiterHelper)
+                    , iterators(it)
+                {
+                    min_ik = INT_MAX;
```

Comment:
Let's replace this by the first int.

Reply:
Should I replace this with the smallest int `INT_MIN`?

Reply:
I mean the first integer in the tuple. Numeric limits might not work for BigInts and so on.

Reply:
Oh, but that would again cause us to dereference the end iterator to get the first IK.